src/coffee/order/CoffeeOrderBoard.java

```diff
@@ -0,0 +1,50 @@
+package coffee.order;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CoffeeOrderBoard {
+    private List<Order> orderQueue;
+
+    public CoffeeOrderBoard() {
+        orderQueue = new ArrayList<>();
+    }
+
+    public void add(String customerName) {
+        Order newOrder = new Order(customerName);
+        orderQueue.add(newOrder);
+    }
+
+    public void deliver() {
+        if (!orderQueue.isEmpty()) {
+            Order nextOrder = orderQueue.remove(0);
+            System.out.println("Delivering order: " + nextOrder.getCustomerName());
+        } else {
+            System.out.println("No orders in the queue.");
+        }
+    }
+
+    public void deliver(int orderNumber) {
+        for (int i = 0; i < orderQueue.size(); i++) {
+            Order order = orderQueue.get(i);
+            if (order.getOrderNumber() == orderNumber) {
+                System.out.println("Delivering order: " + order.getCustomerName());
+                orderQueue.remove(i);
+                return;
+            }
+        }
```

Comment:
this method works incorrectly, we need to deliver based on order number not on an index where the order is stored